path_roles.go

```diff
@@ -351,14 +459,38 @@ func (b *azureSecretBackend) pathRoleList(ctx context.Context, req *logical.Requ
 }
 
 func (b *azureSecretBackend) pathRoleDelete(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	resp := new(logical.Response)
+
 	name := d.Get("name").(string)
+	role, err := getRole(ctx, name, req.Storage)
+	if err != nil {
+		return nil, fmt.Errorf("error getting role: %w", err)
+	}
+
+	if role != nil && role.PersistApp {
+		c, err := b.getClient(ctx, req.Storage)
+		if err != nil {
+			return nil, fmt.Errorf("error during delete: %w", err)
+		}
+
+		// unassigning roles and removing group membership is effectively a garbage collection operation.
+		// Errors will be noted but won't fail the revocation process. 
+		// Deleting the app, however, *is* required to consider the secret revoked.
+		if err := removeRolesAndGroupMembership(ctx, c, role); err != nil {
+			resp.AddWarning(err.Error())
```

Comment:
We want to adhere to the style of returning any errors during execution, and we also want to return `nil` as the response from the Delete function instead of an empty `logical.Response` object even when everything executes correctly. In this case I would suggest changing this to be similar to what you have for the other errors above:

```suggestion
			return nil, fmt.Errorf("error during role and group membership cleanup, err=%w", err)
```

Reply:
So removing roles and group membership is a nice to have, but isn't required, only the deletion of the app.  This follows the way this is currently done in the [none persisted app](https://github.com/hashicorp/vault-plugin-secrets-azure/blob/7f4e89517c103485b29d75ca7906fb3e6652f5ee/path_service_principal.go#L250).  As long as the app is deleted these bindings don't matter.  I can still return the error, just want to call out that it might result in deletes faililng more often since we are making more request to Azure.